codec-http2/src/main/java/io/netty/handler/codec/http2/StreamByteDistributor.java

```diff
@@ -46,11 +46,14 @@
         boolean hasFrame();
 
         /**
-         * Determine if a write operation is allowed for this stream. This will typically take into account the
-         * stream's flow controller being non-negative.
-         * @return {@code true} if a write is allowed on this stream. {@code false} otherwise.
+         * The primary driver to determine allocation amount is {@link #streamableBytes()}. However a
+         * {@link StreamByteDistributor} needs to know the stream's window size in order to avoid allocating bytes if
+         * the window size is negative. The window size being {@code 0} may also be significant to determine when if an
+         * stream has been given a chance to write an empty frame, and also enables optimizations like not writing
+         * empty frames in some situations (don't write headers until data can also be written).
+         * @return the size of the stream's flow control window.
          */
-        boolean isWriteAllowed();
+        int windowSize();
```

Comment:
default methods would be nice for this ... let me see what this looks like because the distributors will need to calculate this value anyways.